File: backend/src/main/java/com/ptit/expensetracker/backend/ai/dto/context/FinancialContextDto.java
package com.ptit.expensetracker.backend.ai.dto.context;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;
import java.util.Map;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class FinancialContextDto {
    private List<WalletDto> wallets;
    private List<TransactionDto> recentTransactions;
    private Map<String, Double> categorySpending;
    private List<MonthlyTotalDto> monthlyTotals;
    private Map<String, Object> analyticsCache;

    private Double totalBalance;
    private Double monthlyAvgIncome;
    private Double monthlyAvgExpense;
    private Double savingsRate;
}

File: app/src/main/java/com/uet/expensetracker/features/ai/data/remote/dto/context/FinancialContextDto.kt
package com.uet.expensetracker.features.ai.data.remote.dto.context

data class FinancialContextDto(
    val wallets: List<WalletDto> = emptyList(),
    val recentTransactions: List<TransactionDto> = emptyList(),
    val categorySpending: Map<String, Double> = emptyMap(),
    val monthlyTotals: List<MonthlyTotalDto> = emptyList(),
    val analyticsCache: Map<String, Any?> = emptyMap(),
    val totalBalance: Double? = null,
    val monthlyAvgIncome: Double? = null,
    val monthlyAvgExpense: Double? = null,
    val savingsRate: Double? = null
)

